src/base/portal.tsx

```diff
@@ -18,42 +35,49 @@ export function PortalChild({
 }: {
   children: React.ReactNode;
   renderTo?: PortalPropT;
-  menuSurfaceDomPositionRef?: React.MutableRefObject<HTMLDivElement | null>;
+  menuSurfaceDomPositionRef?: MutableRefObject<HTMLDivElement | null>;
 }) {
   const [portalEl, setPortalEl] = useState<Element | undefined>();
+  const { portalElement: portalElementFromContext } = useContext(PortalContext);
 
   useEffect(() => {
     let element: Element | undefined = undefined;
 
-    if (renderTo === true) {
+    if (renderTo === true && portalElementFromContext) {
```

Comment:
Do you think there is a way to test the portal from context in `index.spec.tsx`? Would be great to verify that it works in scenarios with a portal context element and without.

Reply:
I'm still new to writing tests, but let me see what I can put together! 😄

Reply:
Try your best. Feel free to contact me on Discord if you need some help with writing tests!

Reply:
I added a test for each combination of using PortalProvider and renderToPortal (4 tests). Hopefully these tests are what we're looking for.

I couldn't figure out how to get a reference of the Portal element without somehow modifying the Portal element, so I went ahead and defined an interface named `PortalProps` so that I could pass in a data-attribute. You'll see that as a new change `portal.tsx`.

I was expecting to have to trigger the dialog to open in order for it to be rendered in the portal, but the dialog's `open` property had no effect on when the element was rendered in the DOM, just whether it was shown. I assumed this is expected and removed the user event triggering the dialog to open.

Reply:
@EmiBemi can you take a look at this and see if the tests are alright?